daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/dseltotql/DselToTqlVisitor.java

```diff
@@ -0,0 +1,262 @@
+        }
+    }
+
+    @Override
+    public Expression visitNot(ELNode elNode) {
+        return TqlBuilder.not(elNode.getChild(0).accept(this));
+    }
+
+    @Override
+    public Expression visitHPath(ELNode elNode) {
+        throw new IllegalStateException("Unsupported HPath: " + elNode.getImage());
+    }
+
+    @Override
+    public Expression visitVariable(ELNode elNode) {
+        /*
+         * DSEL supports this since V1.1.0 but not TQL
+         */
+        throw new IllegalStateException("Unsupported Variable: " + elNode.getImage());
+    }
+
+    @Override
+    public Expression visitFunctionCall(ELNode elNode) {
+        final String fieldName = elNode.getChild(0).getImage();
+
+        switch (elNode.getImage()) {
+        case IS_EMPTY:
+            return TqlBuilder.isEmpty(fieldName);
+        case IS_NULL:
+            return TqlBuilder.isNull(fieldName);
+        case MATCHES:
+            return TqlBuilder.match(fieldName, elNode.getChild(1).getImage());
+        case CONTAINS:
+            final String value = elNode.getChild(1).getImage();
+            final ELNode isCaseSensitiveNode = elNode.getFirstChild(ELNodeType.BOOLEAN_LITERAL);
+            return (isCaseSensitiveNode != null && "false".equals(isCaseSensitiveNode.getImage()))
+                    ? TqlBuilder.containsIgnoreCase(fieldName, value)
+                    : TqlBuilder.contains(fieldName, value);
+        case BETWEEN: {
+            return visitBetweenFunction(elNode);
+        }
+        case IN: {
+            return visitInFunction(elNode);
+        }
+        default:
+            throw new IllegalStateException("Unsupported function: " + elNode.getImage());
+        }
+    }
+
+    @Override
+    public Expression visitAssignment(ELNode elNode) {
+        /*
+         * `field_name=123` is considered as an assignment, despite this is the following which should be an assignment:
+         * `let field_name=123`. Finally we decided to consider this case as an equality.
+         */
+        return TqlBuilder.eq(elNode.getChild(0).getImage(), elNode.getChild(1).getImage());
+    }
+
+    @Override
+    public Expression visitIfThenElse(ELNode elNode) {
+        /*
+         * DSEL supports this since V1.1.0 but not yet TQL
+         */
+        throw new IllegalStateException("Unsupported conditional expression : " + elNode.getImage());
+    }
+
+    @Override
+    public Expression visitSwitchCases(ELNode elNode) {
+        /*
+         * DSEL supports this since V1.1.0 but not TQL
+         */
+        throw new IllegalStateException("Unsupported conditional expression : " + elNode.getImage());
+    }
+
+    private Expression[] buildExpressionsArray(ELNode elNode) {
+        final Expression[] expressions = new Expression[elNode.getChildren().size()];
+        IntStream.range(0, elNode.getChildren().size()).forEach(i -> expressions[i] = elNode.getChildren().get(i).accept(this));
+        return expressions;
```

Comment:
_Nitpick_
This is a bit hard to read, you can simplify it like this:

```suggestion
        return elNode.getChildren()
                .stream()
                .map(child -> child.accept(this))
                .toArray(Expression[]::new);
```